draft-ietf-quic-recovery.md

```diff
@@ -538,9 +539,9 @@ max_ack_delay:
   RTT sample less than min_rtt.
 
 reordering_threshold:
-: The largest delta between the largest acked
-  retransmittable packet and a packet containing retransmittable frames before
-  it's declared lost.
+: The largest packet number gap between the largest acked
+  congestion controlled packet and smaller unacknowledged congestion
+  controlled packet before it is declared lost.
```

Comment:
Why does `reordering_threshold` care about congestion-controlled or not? If you have the following outstanding packets:

1. ACK only
2. ACK only
3. ACK only
4. MAX_DATA only

And then you get an ACK for packet 4, should you not consider any of the previous packets as lost? I understand that marking a non-congestion controlled packet as lost might have no effect, but it is lost nonetheless, right?

Reply:
@nibanks You don't even save packets 1 to 3 (in the sent_packets), so when receiving the ACK for 4, you wouldn't even remember that you ever sent them.

Reply:
Yes, I agree you don't *need* to save them, but if you want to, for purely a statistical measurement of loss, couldn't you remember them?

Reply:
You *could* do that, but you'll probably have to be very careful about loss at the tails. I'm not sure if this would be an improvement over a measurement that just uses retransmittable packets though.

Reply:
How would you deal with optimistic ACK attack if you forget? It is valid to ACK an unackable packet if it is together with other packets, right?

Reply:
All packets except Reset are ackable, and so they are included in the ACK when it's sent.  ACK-only packets don't provoke an ACK in response, however.

Reply:
@ianswett What are your thoughts here? Shouldn't all packets, congestion controlled or not, be capable of being lost and therefore be able to be detected as lost?

Reply:
This is one of the great unsolved mysteries of the universe - is information capable of getting lost.